app/services/story_query_builder.rb

```diff
@@ -0,0 +1,192 @@
+require 'elasticsearch/dsl'
+
+class StoryQueryBuilder
+  include Elasticsearch::DSL
+
+  attr_reader :current_user, :params, :query_str, :fields
+
+  def initialize(args)
+    @query_str = args[:query]
+    @fields = args[:fields] || default_fields
+    @current_user = args[:current_user]
+    @params = args[:params]
+
+    build_dsl
+  end
+
+  def to_hash
+    @dsl.to_hash
+  end
+
+  def as_dsl
+    @dsl
+  end
+
+  def default_operator
+    return 'and' unless params && params[:operator]
+    params[:operator]
+  end
+
+  def default_fields
+    ['title', 'short_description', 'description']
+  end
+
+  def apply_authz?
+    current_user
+  end
+
+  def apply_published_filter?
+    apply_authz?
+  end
+
+  def composite_query_string
+    stringify_clauses [query_str].select(&:present?)
+  end
+
+  def humanized_query_string
+    stringify_clauses [query_str].select(&:present?)
+  end
+
+  private
+
+  def stringify_clauses(clauses)
+    if clauses.length == 2
+      clauses.map { |c| "(#{c})" }.join(" AND ")
+    elsif clauses.length == 1
+      clauses[0].to_s
+    else
+      ""
+    end
+  end
+
+  def munge_fielded_params(fielded)
```

Comment:
What is this used for?  Some sort of created-in-range query that we don't use yet?

Reply:
yes. I basically lifted a query builder class I built a couple years ago and started adapting. There are several vestigial pieces. I will mark those so it's clear what isn't (yet) used.